### PR TITLE
BUG: Fixes Incompatible return value type in pandas/core/generic.py

### PR DESCRIPTION
**Pull request by bunardsheng**
- Contributes to issue #37715 (https://github.com/pandas-dev/pandas/issues/37715)
- Casts 64-bit signed integer to int value
- Passes unit tests through PyTest
- Added tests to respective folders within tests/series and tests/frame subdirectory
- Removed ignore comments

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -658,7 +658,7 @@ def size(self) -> int:
         """
         # error: Incompatible return value type (got "signedinteger[_64Bit]",
         # expected "int")  [return-value]
-        return np.prod(self.shape)  # type: ignore[return-value]
+        return int(np.prod(self.shape))  # type: ignore[return-value]
```
Comment (twoertwein) [inline pandas/core/generic.py:661]: Now it should be possible to remove the error&ignore comment, or?

**Review by rhshadrach — CHANGES_REQUESTED** [review]
While subtle, this changes from a NumPy int to a Python int. I think this should be noted in the whatsnew as a bugfix.

Are the changes to pandas_web intentional?

**Comment by bunardsheng** [comment]
> While subtle, this changes from a NumPy int to a Python int. I think this should be noted in the whatsnew as a bugfix.
> 
> Are the changes to pandas_web intentional?

My mistake, I will revert the changes as the changes in web pertain to a separate issue.

File: pandas/_testing/size_tests.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import numpy as np
+
+import pandas as pd
+
+
+def test_size():
```
Comment (rhshadrach) [inline pandas/_testing/size_tests.py:6]: `_testing` is for utility functions used in the tests. Can you move this to `pandas.tests`.

Comment (bunardsheng) [reply pandas/_testing/size_tests.py:6]: I have moved the testing to the pandas tests/generic folder, thank you for the comment!

Comment (rhshadrach) [reply pandas/_testing/size_tests.py:6]: Looks like this test still needs to be removed from this file. Also, I think it should be placed in `tests/frame/methods/test_size.py` and `tests/series/methods/test_size.py`. These will be a new files (they don't exist yet) and require breaking up the test into one for Series and one for DataFrame.

File: pandas/_testing/size_tests.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    # test for Series object
+    s = pd.Series({"a": 1, "b": 2, "c": 3})
+    assert s.size == 3
+    assert isinstance(s.size, int)
+
+    # test for DataFrame object
+    df = pd.DataFrame({"col1": [1, 2], "col2": [3, 4]})
+    assert df.size == 4
+    assert isinstance(df.size, int)
```
Comment (rhshadrach) [inline pandas/_testing/size_tests.py:15]: Can you parameterize this test instead. I'd recommend something like (but there are other alternatives)

```
@pytest.mark.parametrize("data, index, expected", [({}, None, 0), ({"a": [1], "b": [2]}, ["x"], 2), ...])
```

and then doing `df = pd.DataFrame(data, index)`

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/_testing/size_tests.py (hunk 3 above)
Comment (bunardsheng) [reply pandas/_testing/size_tests.py]: Thanks for the tips. These changes should be pushed by now.

Comment (bunardsheng) [reply pandas/_testing/size_tests.py]: I seem to be running into a few errors with docstring and pre-commit hooks. Are these syntax-related errors within the test files I introduced?

Comment (rhshadrach) [reply pandas/_testing/size_tests.py]: I think this is related to @twoertwein's comment above. If you're not familiar, you can click `Details` within the Job to see what is failing. Here is a direct link:

https://github.com/pandas-dev/pandas/actions/runs/4853717356/jobs/8650205620?pr=52897#step:11:399

File: pandas/tests/frame/methods/test_size.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    [
+        ({"col1": [1], "col2": [3]}, None, 2),
+        ({}, None, 0),
+        ({"col1": [1, np.nan], "col2": [3, 4]}, None, 4),
+        ({"col1": [1, 2], "col2": [3, 4]}, [["a", "b"], [1, 2]], 4),
+        ({"col1": [1, 2, 3, 4], "col2": [3, 4, 5, 6]}, ["x", "y", "a", "b"], 8),
+    ],
+)
+def test_size(data, index, expected):
+    df = DataFrame(data, index=index)
```
Comment (rhshadrach) [inline pandas/tests/frame/methods/test_size.py:18]: Can you add `# GH#52897` as the first line here to indicate when this test was added.

File: pandas/tests/series/methods/test_size.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        ({"a": 1, "b": 2, "c": 3}, None, 3),
+        ([1, 2, 3], ["x", "y", "z"], 3),
+        ([1, 2, 3, 4, 5], ["x", "y", "z", "w", "n"], 5),
+        ([1, 2, 3], None, 3),
+        ([1, 2, 3], ["x", "y", "z"], 3),
+        ([1, 2, 3, 4], ["x", "y", "z", "w"], 4),
+    ],
+)
+def test_series(data, index, expected):
+    s = Series(data, index=index)
```
Comment (rhshadrach) [inline pandas/tests/series/methods/test_size.py:19]: Same thing here - `# GH#52897`

Comment (rhshadrach) [reply pandas/tests/series/methods/test_size.py:19]: Can you use `ser` instead - as a general rule we stay away from 1 letter variables to help with grepping.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looks good! A couple of nitpicks and I think this should get an entry in the whatsnew for 2.1.0

File: pandas/tests/series/methods/test_size.py (hunk 5 above)
Comment (bunardsheng) [reply pandas/tests/series/methods/test_size.py]: Thank you for the tips, these changes should be reflected in the most recent commit.

File: pandas/tests/series/methods/test_size.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        ([1, 2, 3, 4, 5], ["x", "y", "z", "w", "n"], 5),
+        ([1, 2, 3], None, 3),
+        ([1, 2, 3], ["x", "y", "z"], 3),
+        ([1, 2, 3, 4], ["x", "y", "z", "w"], 4),
+    ],
+)
+
+# GH#52897
+def test_series(data, index, expected):
+    ser = Series(data, index=index)
```
Comment (rhshadrach) [inline pandas/tests/series/methods/test_size.py:21]: Ah, sorry - the comment should be the first line of the test. Nothing wrong with this really, but I do think we should be consistent with the style of the other tests.

```
def test_series(data, index, expected):
    # GH#52897
    ser = Series(data, index=index)
```

Same goes for the other test as well.

File: pandas/core/generic.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/generic.py:661]: Just a reminder on this comment - you should be able to remove the `type: ignore` comment (and the comment above this line)

File: pandas/tests/series/methods/test_size.py (hunk 6 above)
Comment (bunardsheng) [reply pandas/tests/series/methods/test_size.py]: Thanks for the note! All changes should be pushed.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
All looks great - I think it just needs a note in the whatsnew under `Bug fixes` in the `Numeric` section.

**Comment by bunardsheng** [comment]
> All looks great - I think it just needs a note in the whatsnew under `Bug fixes` in the `Numeric` section.

The entry has been added to the whatsnew.

**Review by twoertwein — APPROVED** [review]
I like the change :)

Even though it is a small change it might break code for people who explicitly converted it to `int` using `Series.size.item()`. I hope this doesn't require a deprecation cycle @jbrockmendel

**Comment by jbrockmendel** [comment]
No objections here

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
@bunardsheng - conflict needs to be resolved and we're good to go.

**Comment by bunardsheng** [comment]
> @bunardsheng - conflict needs to be resolved and we're good to go.

All conflicts should be resolved. Can this PR be merged now, and should I close the PR?

**Comment by rhshadrach** [comment]
Thanks @bunardsheng!

**Pull request merged**